### PR TITLE
feat(apmgrpc): wrap the server-stream with transaction-ctx

### LINKED ISSUE #1148: Improve gRPC Stream Interceptor
**Is your feature request related to a problem? Please describe.**
The `NewStreamServerInterceptor` creates a Transaction, but does not pass this transaction on to the handler.
This means that I can not create spans inside my gRPC handler and make them attached to the transaction created by the interceptor.

**Describe the solution you'd like**
I'd like `apmgrpc` to wrap the gRPC server-stream with a struct that returns the `context.Contex` it created when creating the transaction, whenever the streams `.Context()` method is called. This is how it is done in many other stream-interceptors, such as in this logrus-interceptor: https://github.com/grpc-ecosystem/go-grpc-middleware/blob/master/logging/logrus/server_interceptors.go#L58-L61

**Describe alternatives you've considered**
Current workarounds:
- Drop using a streaming-interceptor, and add parsing of trace-headers from metadata and create transactions myself in every streaming call.
- Drop using apmgrpc's streaming-interceptor, but blatantly copy it and edit it to add the context-wrapping lines myself.

### PR DESCRIPTION
**Pull request by bendiktv2**
This allows the handler to retrieve the transaction from it's context, to e.g. create spans.

This closes #1148

### CODE REVIEW

**Review by stuartnelson3 — COMMENTED** [review]
Looks good to me, but I'll wait for @axw for final approval. He'll be back next week.

File: module/apmgrpc/server.go
Comment (axw) [inline module/apmgrpc/server.go:40]: go-grpc-middleware is currently only used in tests, and I'd like to keep it that way to minimise dependencies added to instrumented applications.

Can you please remove it and introduce a type similar to WrappedServerStream (but unexported)?

**Review by axw — CHANGES_REQUESTED** [review]
Thanks for the PR! The approach looks good, I'd just like to avoid the dependency please.

File: module/apmgrpc/server.go
Comment (bendiktv2) [reply module/apmgrpc/server.go]: Sure, I used it, since it was already imported, but have now updated the PR without the import. I also removed the check for if the stream was already a wrappedServerStream, since it's no longer needed when it's not exported.

**Review by axw — APPROVED** [review]
Thank you :)

**Pull request merged**
